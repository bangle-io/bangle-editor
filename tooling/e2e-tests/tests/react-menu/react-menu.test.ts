import { expect, Page, test } from '@playwright/test';
import path from 'path';

import {
  ctrlKey,
  mountEditor,
  pmRoot,
  pressLeft,
  sleep,
} from '../../setup/helpers';

test.beforeEach(async ({ page, baseURL }, testInfo) => {
  const url = `${baseURL}/${path.basename(__dirname)}`;
  await page.goto(url, { waitUntil: 'networkidle' });
});

const isMenuActive = (page: Page) =>
  page.evaluate(() => {
    const win = window as any;
    const view = win.editor.view;
    return win.commands.floatingMenu.queryIsMenuActive(win.floatMenuKey)(
      view.state,
      view.dispatch,
      view,
    );
  });

const insertSticker = (page: Page) =>
  page.evaluate(() => {
    const win = window as any;
    return win.dispatcher(win.commands.sticker.insertSticker('triceratops'));
  });

test.describe('react-menu test', () => {
  test.beforeEach(async ({ page }) => {
    await page.setViewportSize({ width: 600, height: 400 });
    await mountEditor(page);
    await page.keyboard.down(ctrlKey);
    await page.keyboard.press('a');
    await page.keyboard.up(ctrlKey);
    await page.keyboard.press('Backspace', { delay: 10 });
  });

  test('Shows and hides a tooltip menu when arrow keys moved in todo-list', async ({
    page,
  }) => {
    await page.type(pmRoot, '[ ] my task');

    await pressLeft(page, { withShift: true });

    expect(await isMenuActive(page)).toBe(true);

    await expect(page.locator('.bangle-tooltip')).toBeVisible();

    expect(await page.screenshot()).toMatchSnapshot();

    await pressLeft(page, { withShift: false });

    expect(await isMenuActive(page)).toBe(false);
  });

  test('menu positions correctly for NodeSelections', async ({ page }) => {
    await page.type(pmRoot, '[ ] my task');
    await insertSticker(page);
    await pressLeft(page);

    expect(await isMenuActive(page)).toBe(true);

    await expect(page.locator('.bangle-tooltip')).toBeVisible();
  });
});
