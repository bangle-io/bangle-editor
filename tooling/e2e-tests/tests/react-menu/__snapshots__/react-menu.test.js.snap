// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-menu test Shows and hides a tooltip menu when arrow keys moved in todo-list 1`] = `"<div class=\\"bangle-tooltip-content\\"></div>"`;

exports[`react-menu test menu positions correctly for NodeSelections 1`] = `"<div class=\\"bangle-tooltip-content\\"></div>"`;
